src/berry_mill/kiwrap.py

```diff
@@ -85,6 +87,17 @@ def add_repo(self, reponame: str, repodata: Dict[str, str]) -> KiwiParent:
 
         return self
 
+    def _verify_gpg_key(self, key_path) -> bool:
+        """
+        Verify wether the downloaded file is a GPG key
+        """
+        try:
+            return bool(subprocess.run(['gpg', '--dearmor', key_path], capture_output=True, text=True).returncode == os.EX_OK)
+
+        except Exception as e:
+            log.warning(f"An error occurred: {e}")
+            return False
+
     def _get_repokeys(self, reponame: str, repodata: Dict[str, str]) -> str:
```

Comment:
IMO, I would change that to:
```python
def _get_repokeys(self, ....) -> str|None:
```

So then:
```python
a = self._get_repokeys()
if a is not None:
  ...
```

Second, the whole method should be done like so:

```python
g_path: str|None = None

# do somthing to g_path or not
if response.status_code = HTTPStatus.OK:
    with open....

# Separate that check and do it _anyway_:
if g_path is not None and not self._verify_gpg_key(g_path):
    g_path = None

return g_path
```

I would prefer not to define `g_path` first to something bogus and then post-modify it, as well I would prefer returning `None` instead of an empty String.

Reply:
Yes thank you, this is pretty much what I meant with _"if you make the calling functions except a null return value of this function"_ :+1:

Reply:
done